cpp/src/io/orc/stripe_enc.cu

```diff
@@ -408,8 +408,8 @@ static __device__ uint32_t IntegerRLE(
       } else {
         intrle_minmax(vmax, vmin);
       }
-      vmin = warp_reduce(temp_storage.full[0]).Reduce(vmin, cub::Min());
-      vmax = warp_reduce(temp_storage.full[1]).Reduce(vmax, cub::Max());
+      vmin = warp_reduce(temp_storage.full[t / 32]).Reduce(vmin, cub::Min());
+      vmax = warp_reduce(temp_storage.full[(t + block_size) / 32]).Reduce(vmax, cub::Max());
```

Comment:
does the half warp reduce temp storage also need updating?

Reply:
should this be `(block_size) / 2 + t / 32` ?

Reply:
half storage did not need to be updated, I'll revert that

Reply:
uh... actually why wouldn't we reuse the tmp storage? The two reductions are not happening in parallel.

Reply:
@vuule I think you cannot assume that threads in a warp will be in lock step. Either we allocate two separate shared memory locations for warp reduce or we use syncwarp so that two reduction process don't access the same location.

Reply:
I was under the impression that threads within a warp are logically in sync. Is that false?